src/ess.jl

```diff
@@ -255,37 +255,39 @@ function ess_rhat(
 )
     # compute size of matrices (each chain may be split!)
     niter = size(chains, 1) ÷ split_chains
-    nparams = size(chains, 3)
     nchains = split_chains * size(chains, 2)
     ntotal = niter * nchains
 
+    # vector used to construct output vector types
+    params = view(chains, firstindex(chains, 1), firstindex(chains, 2), :)
+
```

Comment:
How do you expect the `view` could be problematic? Since we're using `eachslice`, we're already using views of this array.

Reply:
But is that the case just because DimensionalData and AxisArrays do not implement `similar` correctly (or, rather, not all methods that would be needed)? According to the docstring of `similar` it seems we want to do what this method was designed for:
```julia
 similar(storagetype, axes)

  Create an uninitialized mutable array analogous to that specified by storagetype, but with axes specified by the last argument.

  Examples:

  similar(Array{Int}, axes(A))

  creates an array that "acts like" an Array{Int} (and might indeed be backed by one), but which is indexed identically to A. If A has conventional indexing, this will be identical to Array{Int}(undef,
  size(A)), but if A has unconventional indexing then the indices of the result will match A.

  similar(BitArray, (axes(A, 2),))

  would create a 1-dimensional logical array whose indices match those of the columns of A.
```
That's also how OffsetArratys implements it:
```julia
julia> using OffsetArrays

julia> x = OffsetArray(randn(3, 5), -1:1, 2:6)
3×5 OffsetArray(::Matrix{Float64}, -1:1, 2:6) with eltype Float64 with indices -1:1×2:6:
  0.143167   0.0881142  -1.24245   -0.190666   0.021432
 -0.694397   1.32132    -0.447271  -0.196665   0.46701
  1.59353   -0.0241462   1.54666   -2.07788   -0.403454

julia> similar(x, Missing, (axes(x, 2),))
5-element OffsetArray(::Vector{Missing}, 2:6) with eltype Missing with indices 2:6:
 missing
 missing
 missing
 missing
 missing

julia> axes(similar(x, Missing, (axes(x, 2),)))
(OffsetArrays.IdOffsetRange(values=2:6, indices=2:6),)
```

Reply:
There seems to be an additional (funny) behaviour of AxisArrays:
```julia
julia> using AxisArrays

julia> aa = AxisArray(randn(10, 4, 2));

julia> AxisArrays.axes(aa, 3)
Axis{:page, Base.OneTo{Int64}}(Base.OneTo(2))

julia> similar(aa, Int, (AxisArrays.axes(aa, 3),))
3-dimensional AxisArray{Int64,3,...} with axes:
    :row, Base.OneTo(10)
    :col, Base.OneTo(4)
    :page, Base.OneTo(2)
And data, a 10×4×2 Array{Int64, 3}:
[:, :, 1] =
 140405193817504  140405014786992  140405014786992  140405014786992
               0                1                1                1
 140405014758288  140405014786992  140405014786992  140405014786992
               0                1                1                1
 140405201715984  140405014786992  140405014786992  140405014786992
               0                1                1                1
 140405014786992  140405014786992  140405014786992  140405014786992
               1                1                1                1
 140405014786992  140405014786992  140405014786992  140405014786992
               1                1                1                1

[:, :, 2] =
 140405014786992  140405014786992  140405014786992  140405014786992
               1                1                1                1
 140405014786992  140405014786992  140405014786992  140405014786992
               1                1                1                1
 140405014786992  140405014786992  140405014786992  140405014786992
               1                1                1                1
 140405014786992  140405015840704  140405014786992  140405014786992
               1                0                1                1
 140405014786992  140405201715984  140405014786992                0
               1                0                1                0
```

Reply:
Okay, I see what you mean now, and I agree that this should be supportable by DimensionalData (see https://github.com/rafaqz/DimensionalData.jl/issues/437). I'll make the changes here.